OHHTTPStubs.podspec

```diff
@@ -33,8 +33,9 @@ Pod::Spec.new do |s|
   s.watchos.deployment_target = '2.0'
   s.tvos.deployment_target = '9.0'
 
-  s.default_subspec = 'Default'
+  s.swift_version = '5.0'
```

Comment:
Just realized, that might mean that CocoaPods will refuse to install the pod in a 4.x project even though it's compatible…

Maybe until we use the range attribute when next CP is out of beta, we should use a lower value here?

Worth manually checking with a project still using Xcode 9 or 10.0 and Swift 4 at least

Reply:
I think you're right that this parameter is going to cause problems.

## Setting `swift_version` to 5.0

When the `.podspec` has `swift_version` set to 5.0 and I have set the Swift Example project to use Swift 4.0, I can't compile the project with Xcode 10.1.

<img width="422" alt="Screen Shot 2019-04-10 at 8 40 26 PM" src="https://user-images.githubusercontent.com/2142301/55923038-c43aef80-5bd1-11e9-8a8b-39a7e38ee44a.png">

A `pod install` or `pod update` will update the `Pods` project to use Swift 4.2, which is then overridden by the OHHTTPStubs target.  

<img width="396" alt="Screen Shot 2019-04-10 at 8 44 30 PM" src="https://user-images.githubusercontent.com/2142301/55923111-124ff300-5bd2-11e9-8353-f37696f131c5.png">

This could be fixed in Swift 4.0 projects by requiring a `post_install` step in Podfiles, something along the lines of 

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    if ['OHHTTPStubs'].include? target.name
      target.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '4.0'
      end
    end
  end
end
```

We could update the README with this, but it would just add extra work to anyone updating or installing the project, and could result in frustration.  

### Removing `swift_version`

If we remove `swift_version` from the `.podspec`, Cocoapods will use whatever swift version the compiled app is set to use.

Here's Xcode 10.1 and Swift 4, after `pod install`:

<img width="395" alt="Screen Shot 2019-04-10 at 9 00 22 PM" src="https://user-images.githubusercontent.com/2142301/55923605-2a287680-5bd4-11e9-8db3-f1834d6c7d42.png">

And here's Xcode 10.2 and Swift 5 after `pod install`:

<img width="383" alt="Screen Shot 2019-04-10 at 9 02 55 PM" src="https://user-images.githubusercontent.com/2142301/55923608-2bf23a00-5bd4-11e9-9967-ae7eb4190374.png">

 

After all of this, I think we should remove `swift_version` from the `.podspec`, do a release for Swift 5.0, then we can add the range in the next release when CocoaPods 1.7.0 is fully released.

Reply:
What a detailed investigation 😅
Yes I agree until CP 1.7 is out and the attribute accepting a range of versions is supported, we should remove the swift_version attribute from the podspec then.

Reply:
Any plan for merging this soon? Seems like it is ready to be deployed 😬.
My project is fully migrated to Swift 5, just waiting for this release in order to re-enable my network stubs for unit testing purposes.

Reply:
@eMdOS as you can see from the conversation above, merging this as-is would mean breaking the backward compatibility with older Xcode and study versions, so we need to address that first by fixing the podspec, only then child we release 😅

Reply:
I think this one's ready to roll now @AliSoftware .

Reply:
@Alisoftware - friendly bump. :)